internal/debug/debug.go

```diff
@@ -0,0 +1,59 @@
+package debug
+
+import (
+	"context"
+	"net/http"
+	"net/http/pprof"
+	"time"
+
+	"github.com/CRED-CLUB/propeller/pkg/logger"
+	"github.com/gorilla/mux"
+)
+
+// DebugServer holds the rest server with debug endpoints
+type DebugServer struct {
+	url    string
+	router *mux.Router
+	server *http.Server
+}
+
+// NewDebugServer creates a new instance of debug server
+func NewDebugServer(URL string) *DebugServer {
+	router := mux.NewRouter()
+	srv := &http.Server{Addr: URL, Handler: router}
+	return &DebugServer{
+		url:    URL,
+		router: router,
+		server: srv,
+	}
+}
+
+// Start starts the rest server
+func (ds *DebugServer) Start() {
+	logger.Log.Info("Starting debug server")
+	ds.registerDebugRoutes()
+	go func() {
+		err := ds.server.ListenAndServe()
+		if err != nil {
+			logger.Log.Errorf(err.Error())
+		}
+	}()
+}
+
+// Stop the rest server
+func (ds *DebugServer) Stop(pctx context.Context) {
+	ctx, cancel := context.WithTimeout(pctx, 5*time.Second)
+	defer cancel()
+	ds.server.Shutdown(ctx)
+}
+
+// registerDebugRoutes registers all the debug routes
+func (ds *DebugServer) registerDebugRoutes() {
```

Comment:
is it possible to put registering these debug routes behind a config to enable profiling? The config file is defined at: https://github.com/CRED-CLUB/propeller/blob/main/config/propeller.toml

Reply:
These routers are static. I feel the config file should be used to drive user provided parameters. I have pushed new changes that won't require registering routes now.